functions/helpers.bash

```diff
@@ -558,11 +558,148 @@ set_cpu_speed() {
 }
 
 
-## check if InfluxDB is installed
+## Functions to check it a given supported program is installed should be placed below here
+
+## Check if amanda is installed
+##
+##    amanda_is_installed
+##
+amanda_is_installed() {
+  if dpkg -s 'amanda-common' 'amanda-server' 'amanda-client' &> /dev/null; then return 0; fi
+  return 1
+}
+
+## Check if InfluxDB is installed
 ##
-##    iinfluxdb_is_installed
+##    influxdb_is_installed
 ##
 influxdb_is_installed() {
-  if [[ -d /var/lib/influxdb ]]; then return 0; fi
+  if dpkg -s 'influxdb' &> /dev/null && [[ -d /var/lib/influxdb ]]; then return 0; fi
+  return 1
+}
+
+## Check if Grafana is installed
+##
+##    grafana_is_installed
+##
+grafana_is_installed() {
+  if dpkg -s 'grafana' &> /dev/null && [[ -s /etc/grafana/grafana.ini ]]; then return 0; fi
+  return 1
+}
+
+## Check if node is installed
+##
+##    node_is_installed
+##
+node_is_installed() {
+  if [[ -x $(command -v npm) ]] || [[ $(node --version) == "v14"* ]]; then return 0; fi
+  return 1
+}
+
+## Check if samba is installed
+##
+##    samba_is_installed
+##
+samba_is_installed() {
+  if dpkg -s 'samba' &> /dev/null && [[ -s /etc/samba/smb.conf ]]; then return 0; fi
+  return 1
+}
+
+## Check if firemotd is installed
+##
+##    firemotd_is_installed
+##
+firemotd_is_installed() {
+  if [[ -d /opt/firemotd ]] && [[ -x $(command -v FireMotD) ]]; then return 0; fi
+  return 1
+}
+
+## Check if exim is installed
+##
+##    exim_is_installed
+##
+exim_is_installed() {
+  if dpkg -s 'mailutils' 'exim4' &> /dev/null; then return 0; fi
+  return 1
+}
+
+## Check if homegear is installed
+##
+##    homegear_is_installed
+##
+homegear_is_installed() {
+  if dpkg -s 'homegear' &> /dev/null && [[ -d /var/lib/homegear ]]; then return 0; fi
+  return 1
+}
+
+## Check if mqtt is installed
+##
+##    mqtt_is_installed
+##
+mqtt_is_installed() {
```

Comment:
shouldn't that be mosquitto rather than mqtt

Reply:
I was lazy and didn't want to type it all the way out, plus the install function uses mqtt as well, I can change it if you want

Reply:
yes please

Reply:
fixed